filament/src/Froxelizer.cpp

```diff
@@ -838,43 +811,41 @@ void Froxelizer::froxelizePointAndSpotLight(
     float4 const * const UTILS_RESTRICT boundingSpheres = mBoundingSpheres;
     for (size_t iz = z0 ; iz <= z1; ++iz) {
         float4 cz(s);
-        // froxel that contain the center if ths sphere is special, we don't even need to do the
+        // froxel that contain the center of the sphere is special, we don't even need to do the
         // intersection check, it's always true.
         if (UTILS_LIKELY(iz != zcenter)) {
             cz = spherePlaneIntersection(s, (iz < zcenter) ? planesZ[iz + 1] : planesZ[iz]);
         }
 
         if (cz.w > 0) { // intersection of light with this plane (slice)
-            // the sphere (light) intersects this slice's plane and we now have a new, smaller
+            // the sphere (light) intersects this slice's plane, and we now have a new smaller
             // sphere centered there. Now, find x & y slices that contain the sphere's center
             // (note: this changes with the Z slices)
             const float2 clip = project(p, cz.xyz);
-            const auto indices = clipToIndices(clip);
-            const size_t xcenter = indices.first;
-            const size_t ycenter = indices.second;
+            auto const [xcenter, ycenter] = clipToIndices(clip);
 
             for (size_t iy = y0; iy <= y1; ++iy) {
                 float4 cy(cz);
-                // froxel that contain the center if ths sphere is special, we don't even need to
+                // froxel that contain the center of the sphere is special, we don't even need to
                 // do the intersection check, it's always true.
                 if (UTILS_LIKELY(iy != ycenter)) {
                     float4 const& plane = iy < ycenter ? planesY[iy + 1] : planesY[iy];
-                    cy = spherePlaneIntersection(cz, plane.y, plane.z);
+                    cy = spherePlaneIntersection(cz, plane);
                 }
 
                 if (cy.w > 0) {
-                    // The reduced sphere from the previous stage intersects this horizontal plane
+                    // The reduced sphere from the previous stage intersects this horizontal plane,
                     // and we now have new smaller sphere centered on these two previous planes
                     size_t bx = std::numeric_limits<size_t>::max(); // horizontal begin index
                     size_t ex = 0; // horizontal end index
 
-                    // find the begin index (left side)
-                    for (size_t ix = x0; ix < x1; ++ix) {
+                    // find the "begin" index (left side)
+                    for (size_t ix = x0; ix < x1 + 1; ++ix) {
```

Comment:
I assume that's where the + 1 now is right?

Reply:
yes